src/testing/mod.rs

```diff
@@ -379,15 +377,15 @@ pub trait SystemUnderTest<'a>: Default + Send + Sync {
         let mut keystores = Vec::new();
         for (key_stream, key_pairs, initial_grants) in users {
             let mut rng = ChaChaRng::from_rng(&mut rng).unwrap();
-            let ledger = ledger.clone();
-            let storage = Arc::new(Mutex::new(self.create_storage().await));
-            ledger.lock().await.storage.push(storage.clone());
-
+            let mut loader = TrivialKeystoreLoader {
+                dir: TempDir::new("test_keystore").unwrap().into_path(),
```

Comment:
This will leak the directory for each test we run. We need to instead somehow save the `TempDir` object for the duration of the test and then let it be dropped normally without calling `into_path()`.

One way to handle this would be to have this function return a `Vec<TempDir>`, so the caller can keep the temp dir objects on the stack until it returns. Another option would be to create a struct that wraps a `T: SystemUnderTest` (similar to how we have `MockLedger` that wraps a `MockNetwork` and also provides some common functionality of its own) that has a `Vec<TempDir>` internally which can be appended to whenever we create a new `TempDir`. Or, we could add `new_temp_dir() -> PathBuf` to  the `SystemUnderTest` trait and just require all impls of that trait to keep a `Vec<TempDir>` internally.

Reply:
Decided to just return the `TempDir` in the vector for now.  I think in the long run it makes more sense to put this in a wrapper or in this trait.  I will follow up with that.